.github/workflows/publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish
+on:
+  push:
+      branches: [master]
+
+jobs:
+  build:
+    uses: ./.github/workflows/build-and-test.yml
+  publish:
+    needs: build
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/semversioner
+    permissions:
+        id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install Poetry 1.8.2
+        uses: abatilo/actions-poetry@v3.0.0
+        with:
+          poetry-version: '1.8.2'
+
+      - name: Install dependencies
+        shell: bash
+        run: poetry install
+
+      - name: Build Distributable
+        shell: bash
+        run: poetry build
```

Comment:
I'm not an expert on Github Actions as I normally use other CI/CD tools, but I'd like to find a way that we build only once, and deploy exactly the same artifact we run the tests against. From what I understand from this workflow is that we are building it again, pulling dependencies, etc. The poetry.lock file helps reducing the risk to introducting non tested behaviour, but overall I'd like to have a CI/CD workflow that enforces:

After merging to master:
- We run tests
- We package only once
- We publish the package we run the tests against

Does it make sense? I'll read a bit more about how to achieve that in actions.